README.md

```diff
@@ -3,40 +3,166 @@
+
+### Types
+
+The option `type` can be used to further target transform to a particular type of ember object within the application or addon. The types can be any of the following:
+
+- Services (`--type=services`)
+- Routes (`--type=routes`)
+- Components (`--type=components`)
+- Controllers (`--type=controllers`)
+
+The path of the file being transformed is matched against the glob pattern of the type to make the decision on whether to proceed with transforms.
+
+If a type is not provided, the codemods will run against all the files in the path provided.
+
+### Runtime Config Path
+
+The codemods convert the code from one API to another by statically analyzing the code. While this works well for most of the times, there are some requirements which need little more insights into the code to figure out the best way to transform. For example, if we need to determine the class hierarchy, it is not possible with static analysis to determine the super classes and their properties.
+
+The codemods are designed with `runtime data` as input to correctly transform the code. For each file currently being transformed, the codemods would need a `configuration file` which will provide metadata about the properties of the ember object.
+
+The runtime config path must point to a JSON file containing runtime data of the format:
+
+```
+{
+  data: [{
+     "/absolute/file/path": {
+        "computedProperties": ['computedProp1', ...],
+        "observedProperties": ['observedProp1', ...],
+        "observerProperties": {
+          "observerProp1": ["prop1", "prop2", ...]
+        },
+        "offProperties": {
+          "offProp": ["prop3", ...]
+        },
+        "overriddenActions": ["overriddenAction1", ...],
+        "overriddenProperties": ["overriddenProp1"],
+        "ownProperties": ["prop1", ...],
+        "type": "Component|Route|Controller|EmberObject",
+        "unobservedProperties": {
+          "unobservedProp1": ["prop1", ...]
+        }
+      }
+   }]
+}
+```
+
+_How to get this runtime data from an ember application?_
+
+There are tools available which provides a way of extracting the runtime data. Following are some examples:
+
+- [Dyfactor](https://github.com/dyfactor/dyfactor)
+- [code-to-json](https://github.com/mike-north/code-to-json)
+
+A [dyfactor plugin](https://github.com/ssutar/ember-es6-class-codemod-dyfactor) is developed to extract the runtime data from an ember application. However any tool which provide the runtime data with the expected format can be used with codemods.
+
+## Debugging
+
+The codemods logs the details in the `codemods.log` file in the current directory from where the codemods are being executed. It logs all the details including failures and reasons for failures. It would be the starting point to debug in case of any issues.
```

Comment:
> The codemods logs the details in the `codemods.log` file in the current directory from where the codemods are being executed. It logs all the details including failures and reasons for failures. It would be the starting point to debug in case of any issues.

The codemods log execution information in the `codemods.log` file in the current directory from where the codemods are being executed. Specifically, details such as failures and reasons for failures, are logged. This would be the recommended starting point for debugging issues related to these codemods.